Lib/subprocess.py

```diff
@@ -809,6 +822,40 @@ def __init__(self, args, bufsize=-1, executable=None,
             else:
                 line_buffering = False
 
+        gids = []
+        if groups is not None:
+            if isinstance(groups, str):
+                raise ValueError("Groups must be a list, not a string")
+
+            for group in groups:
+                if isinstance(group, str):
+                    gids.append(grp.getgrnam(group).gr_gid)
+                elif isinstance(group, int):
+                    gids.append(group)
+                else:
+                    raise TypeError("Group must be a string or integer")
+
+            if not gids:
+                raise ValueError("Groups cannot be empty")
+
+            # make sure that the gids are all positive here so we can do less
+            # checking in the C code
+            for gid in gids:
+                if gid < 0:
+                    raise ValueError(f"Group ID cannot be negative, got {gid}")
+
+        uid = None
+        if user is not None:
+            if isinstance(user, str):
+                uid = pwd.getpwnam(user).pw_uid
+            elif isinstance(user, int):
+                uid = user
+            else:
+                raise TypeError("User must be a string or an integer")
+
+            if uid < 0:
```

Comment:
can we feasibly check it against a maximum uid_t value?

Reply:
Not practically in python code, as I mentioned above.

It does look like `PyArg_ParseTuple()` actually does do overflow checking when parsing:
https://github.com/python/cpython/blob/master/Python/getargs.c#L777

So even if an overflow value gets passed in, it will generate an `OverflowError`

Reply:
Presumably the syscall would also generate an error if the kernel had limits other than the normal range of `int`